File: AoC/2025/day01/day01.py
test = [
"L68",
"L30",
"R48",
"L5",
"R60",
"L55",
"L1",
"L99",
"R14",
"L82",
]

def part1():
    start = 50
    zeroes = 0

    turns = read()

    for turn in turns:
        direction = turn[0]
        num = int(turn[1:])
        if direction == "L":
            start -= num
        else:
            start += num

        start %= 100

        if start == 0: zeroes += 1

    return zeroes

def main():
    start = 50
    zeroes = 0

    turns = read()

    for turn in turns:
        direction = turn[0]
        num = int(turn[1:])
        if direction == "L":
            start -= num
        else:
            start += num

        while start < 0:
            start += 100
            zeroes += 1
        while start > 99:
            start -= 100
            zeroes += 1

        if start == 0:
            zeroes += 1

    return zeroes

def part2():
    start = 50
    zeroes = 0
    turns = test
    turns = read()

    for turn in turns:
        num = int(turn[1:])

        if turn[0] == "L":
            num *= -1
            if start == 0:
                start = 100

        start += num
        
        while start < 0:
            start += 100
            zeroes += 1
        while start > 100:
            start -= 100
            zeroes += 1
        if start == 0 or start == 100:
            start = 0 
            zeroes += 1

    return zeroes

def read():
    text = []
    with open(".input") as f:
        for line in f:
            text.append(line.strip())
    return text

if __name__ == "__main__":
    out = part1()
    print(out)

    out = part2()
    print(out) # 6616

    # 6591
    # 6507
    # 6247
    # 6406

File: AoC/2025/day01/day01.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where
    P: AsRef<Path>,
{
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}

fn main() {
    let mut start = 50;
    let mut zeroes = 0;

    if let Ok(lines) = read_lines("./.input") {
        for line in lines.map_while(Result::ok) {
            let (direction, num_str) = line.split_at(1);
            let num: i32 = num_str.parse().expect("Int expected");
            if direction == "L" {
                start -= num;
            } else {
                start += num;
            }
            start %= 100;
            if start == 0 {
                zeroes += 1;
            }
        }
    }
    println!("{}", zeroes);

    start = 50;
    zeroes = 0;

    if let Ok(lines) = read_lines("./.input") {
        for line in lines.map_while(Result::ok) {
            let (direction, num_str) = line.split_at(1);
            let mut num: i32 = num_str.parse().expect("Int expected");
            if direction == "L" {
                num *= -1;
                if start == 0 {
                    start = 100;
                }
            }

            start += num;

            while start < 0 {
                start += 100;
                zeroes += 1;
            }
            while start > 100 {
                start -= 100;
                zeroes += 1;
            }
            if start == 0 || start == 100 {
                start = 0;
                zeroes += 1;
            }
        }
    }
    println!("{}", zeroes);
}
